### PR TITLE
A lot of extra type checks

### PR DESCRIPTION
**Pull request by jespino**
Reduced the number of mypy errors when enabling `disallow_untyped_calls` flag from 388 to 131.

### CODE REVIEW

File: src/consensus/block_header_validation.py [hunk 1]
```diff
@@ -1038,7 +1038,8 @@ def validate_finished_header_block(
                 ip_vdf_iters,
                 header_block.reward_chain_block.infused_challenge_chain_ip_vdf.output,
             )
-            if not header_block.infused_challenge_chain_ip_proof.is_valid(
+
+            if icc_vdf_input is None or not header_block.infused_challenge_chain_ip_proof.is_valid(
```
Comment (jespino) [inline src/consensus/block_header_validation.py:1042]: I don't know if this is 100% correct

File: src/server/address_manager.py [hunk 2]
```diff
@@ -507,13 +507,13 @@ def resolve_tried_collisions_(self):
                         resolved = True
                     elif time.time() - old_info.last_try < 4 * 60 * 60:
                         if time.time() - old_info.last_try > 60:
-                            self.mark_good_(peer, False, time.time())
+                            self.mark_good_(peer, False, math.floor(time.time()))
```
Comment (jespino) [inline src/server/address_manager.py:510]: I'm not 100% sure about the correctness of this, but was used in other place like this.

File: src/server/node_discovery.py [hunk 3]
```diff
@@ -265,10 +267,10 @@ async def _connect_to_peers(self, random):
                     disconnect_after_handshake = True
                     empty_tables = False
                 initiate_connection = self._num_needed_peers() > 0 or has_collision or is_feeler
-                connected = False
+                client_connected = False
```
Comment (jespino) [inline src/server/node_discovery.py:270]: This was reusing a variable name of other type

File: src/server/node_discovery.py [hunk 4]
```diff
@@ -319,11 +321,11 @@ async def _periodically_cleanup(self):
             full_node_connected = self.server.get_full_node_connections()
             connected = [c.get_peer_info() for c in full_node_connected]
             connected = [c for c in connected if c is not None]
-            if len(connected) >= 3:
+            if self.address_manager is not None and len(connected) >= 3:
```
Comment (jespino) [inline src/server/node_discovery.py:324]: Not 100% sure if this is the expected behavior or we prefer something like `assert self.address_manager is not None`

File: src/server/start_service.py [hunk 5]
```diff
@@ -107,13 +109,17 @@ def __init__(
...
-    async def start(self, **kwargs):
+    async def start(self, **kwargs) -> None:
         # we include `kwargs` as a hack for the wallet, which for some
         # reason allows parameters to `_start`. This is serious BRAIN DAMAGE,
         # and should be fixed at some point.
         # TODO: move those parameters to `__init__`
         if self._did_start:
             return
+
+        assert self.self_hostname is not None
```
Comment (jespino) [inline src/server/start_service.py:120]: Not 100% sure if that assertions are going to generate problems, my expectation is the `self_hostname` and the `daemon_port` is stablished before you try to start the service.

File: src/util/block_tools.py [hunk 6]
```diff
@@ -1215,9 +1215,11 @@ def get_icc(
         curr = blocks[curr.prev_hash]
     icc_iters = uint64(vdf_end_total_iters - latest_block.total_iters)
     if latest_block.is_challenge_block(constants):
-        icc_input = ClassgroupElement.get_default_element()
+        icc_input: Optional[ClassgroupElement] = ClassgroupElement.get_default_element()
     else:
         icc_input = latest_block.infused_challenge_vdf_output
+    assert icc_input is not None
```
Comment (jespino) [inline src/util/block_tools.py:1221]: I'm not 100% sure about this assertion.

File: src/wallet/wallet_node.py [hunk 7]
```diff
@@ -459,11 +459,14 @@ async def new_peak_wallet(self, peak: wallet_protocol.NewPeakWallet, peer: WSChi
...
                 self.start_sync()
 
-    def start_sync(self):
+    def start_sync(self) -> None:
         self.log.info("self.sync_event.set()")
         self.sync_event.set()
 
-    async def check_new_peak(self):
+    async def check_new_peak(self) -> None:
+        if self.wallet_state_manager is None:
```
Comment (jespino) [inline src/wallet/wallet_node.py:467]: I think this is the expected behavior becaus is used in other functions in this same file.

**Comment by jespino** [comment]
@mariano54 with this we are half way there to enable the flag. I commented everything that I found more controversial. Everything else is pretty straight forward.

File: src/consensus/block_header_validation.py (hunk 1 above)
Comment (mariano54) [reply src/consensus/block_header_validation.py:1042]: This is correct

File: src/server/address_manager.py (hunk 2 above)
Comment (mariano54) [reply src/server/address_manager.py:510]: Its fine

File: src/server/node_discovery.py (hunk 4 above)
Comment (mariano54) [reply src/server/node_discovery.py:324]: this is fine

File: src/server/start_service.py (hunk 5 above)
Comment (mariano54) [reply src/server/start_service.py:120]: :+1:

**Review by mariano54 — DISMISSED** [review]
Looks good, all the changes you made were correct. Thank you!

**Pull request merged**